widgetssdk/src/main/java/com/glia/widgets/core/secureconversations/SecureConversationsRepository.kt

```diff
@@ -65,6 +69,21 @@ internal class SecureConversationsRepository(private val secureConversations: Se
         }
     }
 
-    fun getUnreadMessagesCount(callback: RequestCallback<Int>) =
-        secureConversations.getUnreadMessageCount(callback)
+    fun getUnreadMessagesCount(callback: RequestCallback<Int>) = secureConversations.getUnreadMessageCount(callback)
+
+    fun getHasPendingSecureConversations(): Single<Boolean> = Single.create { emitter ->
+        secureConversations.getPendingSecureConversationStatus { hasPendingSecureConversations, _ ->
+            if (!emitter.isDisposed) {
+                emitter.onSuccess(hasPendingSecureConversations ?: false)
+            }
+        }
+    }
+
+    /**
+     * Since we're using this function in [EntryWidget] and [EngagementLauncher] where we need the immediate UI response to the user interaction,
+     * it is important to not make the user wait.
+     */
```

Comment:
Let me clarify6 for my understanding. So, when visitor taps 'Contact us' which should show the bottom sheet Entry Widget, if there is ongoing SC, we show the messaging screen after a programmatic 1-sec delay, right? Why is it better than showing it right immediately?

Reply:
No, it will reply with a default value(false) **if the request takes more than 1 second**.

We will have this value right after initialization, so this scenario is a corner case when the internet connection is slow, and EntryWidget is requested right after initialization.

Reply:
Possible improvement: "to not make the user wait" => “to ensure a seamless experience for the user” or “to avoid delays for the user”.
Up to you. 😉